### PR TITLE
Add protocol to the SPI module-info

### PR DESCRIPTION
**Pull request by wendigo**
## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
## Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: core/trino-spi/src/main/java/module-info.java [hunk 1]
```diff
@@ -31,6 +31,7 @@
     exports io.trino.spi.memory;
     exports io.trino.spi.metrics;
     exports io.trino.spi.predicate;
+    exports io.trino.spi.protocol;
```
Comment (martint) [inline core/trino-spi/src/main/java/module-info.java:34]: Why does the SPI have a protocol package?

Comment (wendigo) [reply core/trino-spi/src/main/java/module-info.java:34]: This is for spooling manager SPI. We can move it to some other package if this is the wrong name for it.

Comment (wendigo) [reply core/trino-spi/src/main/java/module-info.java:34]: (SpoolingManager is pluggable)

Comment (martint) [reply core/trino-spi/src/main/java/module-info.java:34]: Yeah, let’s rename the package. That’s doesn’t seem like the right name.

Comment (wendigo) [reply core/trino-spi/src/main/java/module-info.java:34]: What do you propose?

Comment (martint) [reply core/trino-spi/src/main/java/module-info.java:34]: How about io.trino.spi.spool?

File: core/trino-spi/src/main/java/module-info.java [hunk 2]
```diff
@@ -34,6 +34,7 @@
     exports io.trino.spi.resourcegroups;
     exports io.trino.spi.security;
     exports io.trino.spi.session;
+    exports io.trino.spi.spool;
```
Comment (losipiuk) [inline core/trino-spi/src/main/java/module-info.java:37]: should be part of previous commit, right?

**Review by losipiuk — APPROVED** [review]

File: core/trino-spi/src/main/java/module-info.java (hunk 2 above)
Comment (wendigo) [reply core/trino-spi/src/main/java/module-info.java:37]: No. This is a seperate change.

Comment (losipiuk) [reply core/trino-spi/src/main/java/module-info.java:37]: Not really - you should have protocol here before. And rename should influence that. 
Probably just a discussion should you fix a bug before or after rename...

**Pull request merged**